packages/medusa/src/api/routes/store/products/index.ts

```diff
@@ -51,6 +66,47 @@ export const defaultStoreProductsRelations = [
   "type",
 ]
 
+export const defaultStoreProductsFields: (keyof Product)[] = [
+  "id",
+  "title",
+  "subtitle",
+  "status",
+  "external_id",
+  "description",
+  "handle",
+  "is_giftcard",
+  "discountable",
+  "thumbnail",
+  "profile_id",
+  "collection_id",
+  "type_id",
+  "weight",
+  "length",
+  "height",
+  "width",
+  "hs_code",
+  "origin_country",
+  "mid_code",
+  "material",
+  "created_at",
+  "updated_at",
+  "deleted_at",
+  "metadata",
+]
+
+export const allowedStoreProductsFields = [
```

Comment:
Thought: would it make sense that the transformer accept allowed or disallowed? In some cases we want to allow everything except some props or relations props to be accessible. It might be easier in those cases to maintain the constraint. Depending on the output it might be good to include it here and some other places where we put all the props. Wdyt? It would require to slightly update the transformer and util

Cc @olivermrbl

Reply:
Good point, I was thinking the same because some entities can have quite a larger number of `allowedProperties` which could be "pricey" to check.
But then this would be a new concept that we are introducing to controllers and would like to hear what @olivermrbl thinks first :)

Reply:
Would it be best to stick with one or the other to keep a simple API? I like the idea of disallowed properties, as I assume that this is a smaller set than the allowed ones, so with that in mind, it might make sense to switch to that.

Reply:
I think it really depends, if there is no case where we want to allow only a subset of properties then we can go with the disallowed. But if we have both usage, then we will still get stuck a bit but in the opposite way ahah

Reply:
I prefer allowed fields over disallowed. It avoids accidentally leaking things that shouldn't be passed through, and the verbosity is nice and clear. I guess would be that it feels more "pricey" than it actually is so not that concerned about the potential performance hit. 

If we find it is damaging for performance we could consider a different data structure, maybe a set or object where inclusivity checks are O(1)?

Reply:
@srindom the thing with the allowed is that we have to specify all the fields all the way through the last relation
e.g `product -> variant -> [prices, options etc] -> prices -> region -> [tax rate, country etc] -> tax_rate -> [shipping_options, etc] -> shipping_options -> etc etc` it is not a bit too much if the only thing we want to not allowed the access too is on field in on of those relations?